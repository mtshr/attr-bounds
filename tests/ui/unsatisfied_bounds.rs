use attr_bounds::bounds;

#[bounds(T: Copy)]
struct Foo<T>(T);

fn main() {
    let _foo = Foo(Vec::new());
}
