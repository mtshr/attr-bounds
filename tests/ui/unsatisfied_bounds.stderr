error[E0277]: the trait bound `Vec<_>: Copy` is not satisfied
 --> tests/ui/unsatisfied_bounds.rs:7:20
  |
7 |     let _foo = Foo(Vec::new());
  |                --- ^^^^^^^^^^ the trait `Copy` is not implemented for `Vec<_>`
  |                |
  |                required by a bound introduced by this call
  |
note: required by a bound in `Foo`
 --> tests/ui/unsatisfied_bounds.rs:3:13
  |
3 | #[bounds(T: Copy)]
  |             ^^^^ required by this bound in `Foo`
4 | struct Foo<T>(T);
  |        --- required by a bound in this tuple struct

error[E0277]: the trait bound `Vec<_>: Copy` is not satisfied
 --> tests/ui/unsatisfied_bounds.rs:7:16
  |
7 |     let _foo = Foo(Vec::new());
  |                ^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `Vec<_>`
  |
note: required by a bound in `Foo`
 --> tests/ui/unsatisfied_bounds.rs:3:13
  |
3 | #[bounds(T: Copy)]
  |             ^^^^ required by this bound in `Foo`
4 | struct Foo<T>(T);
  |        --- required by a bound in this struct
